tools/manifest-templator/manifest-templator.go

```diff
@@ -116,8 +116,36 @@ func main() {
 		GeneratedManifests: make(map[string]string),
 	}
 
-	if featureGates != nil && strings.Contains(*featureGates, "NonRoot") && !strings.Contains(*featureGates, "NonRootExperimental") {
-		*featureGates = *featureGates + ",NonRootExperimental"
+	if featureGates != nil {
+
+		fgs := strings.Split(*featureGates, ",")
+		newFgs := []string{}
+
+		contains := func(s string, in []string) bool {
+			for _, str := range in {
+				if strings.EqualFold(s, str) {
+					return true
+				}
+			}
+			return false
+		}
+		if contains("NonRoot", fgs) && !contains("NonRootExperimental", fgs) {
+			*featureGates = *featureGates + ",NonRootExperimental"
+		}
+
+		if contains("NonRootExperimental", fgs) {
+			if !contains("NonRoot", fgs) {
+				fgs = append(fgs, "NonRoot")
+			}
+		}
+
+		for _, fg := range fgs {
+			if strings.EqualFold("NonRootExperimental", fg) {
+				continue
+			}
+			newFgs = append(newFgs, fg)
+		}
+		*featureGates = strings.Join(newFgs, ",")
```

Comment:
I have a hard time understanding this block...
Even if `NonRootExperimental` above is added to `fgs` instead of `*featureGates`, why do we add it to ignore it later?
Some explanatory comment(s) would probably also help.

Reply:
After update, isn't this basically doing `*featureGates = strings.Replace(*featureGates, "NonRootExperimental", "NonRoot", 1)`?
If we're worried about case we can probably use a simple regex like `[Nn]on[Rr]oot[Ee]xperimental`.
WDYT?

Reply:
You are right, but it only does it when there is no `NonRoot`.